src/Module.cpp

```diff
@@ -551,20 +551,49 @@ void Module::regdump(uint8_t start, uint8_t len) {
 }
 
 void Module::setRfSwitchPins(RADIOLIB_PIN_TYPE rxEn, RADIOLIB_PIN_TYPE txEn) {
-  _useRfSwitch = true;
-  _rxEn = rxEn;
-  _txEn = txEn;
-  this->pinMode(rxEn, OUTPUT);
-  this->pinMode(txEn, OUTPUT);
+  // This can be on the stack, setRfSwitchTable copies the contents
+  const RADIOLIB_PIN_TYPE pins[] = {
+    rxEn, txEn, RADIOLIB_NC,
+  };
+  // This must be static, since setRfSwitchTable stores a reference.
+  static constexpr RfSwitchMode_t table[] = {
+    {MODE_IDLE,  {LOW,  LOW}},
+    {MODE_RX,    {HIGH, LOW}},
+    {MODE_TX,    {LOW,  HIGH}},
+    END_OF_MODE_TABLE,
+  };
+  setRfSwitchTable(pins, table);
+}
+
+void Module::setRfSwitchTable(const RADIOLIB_PIN_TYPE (&pins)[3], const RfSwitchMode_t table[]) {
+  memcpy(_rfSwitchPins, pins, sizeof(_rfSwitchPins));
+  _rfSwitchTable = table;
+  for (auto pin : _rfSwitchPins)
```

Comment:
I would rather avoid using range loops since that requires C++11. It should be pretty simple to replace this with just a for loop (if number of pins is defined as macro, see one of the next comments)

Reply:
Resolved in 4e26a8a9e0f73f029b3efa3a6301d06bd7d3f29a.